src/Callback.php

```diff
@@ -105,43 +98,29 @@ public function init(): void
      */
     public function set($callback, $args = [])
     {
-        if ($this->postTrigger) {
-            if (isset($_POST[$this->postTrigger])) {
-                $this->app->catch_runaway_callbacks = false;
-                $this->triggered = $_POST[$this->postTrigger];
-
-                $t = $this->app->run_called;
-                $this->app->run_called = true;
-                $ret = call_user_func_array($callback, $args);
-                $this->app->run_called = $t;
-
-                return $ret;
-            }
-        } else {
-            if (isset($_GET[$this->urlTrigger])) {
-                $this->app->catch_runaway_callbacks = false;
-                $this->triggered = $_GET[$this->urlTrigger];
-
-                $t = $this->app->run_called;
-                $this->app->run_called = true;
-                $this->owner->stickyGet($this->urlTrigger);
-                $ret = call_user_func_array($callback, $args);
-                //$this->app->stickyForget($this->name);
-                $this->app->run_called = $t;
-
-                return $ret;
-            }
+        if ($this->isTriggered()) {
+            $this->app->catch_runaway_callbacks = false;
+            $t = $this->app->run_called;
+            $this->app->run_called = true;
+            $ret = call_user_func_array($callback, $args);
+            $this->app->run_called = $t;
+
+            return $ret;
         }
     }
 
     /**
      * Terminate this callback
-     * by rendering the owner view.
+     * by rendering the owner view by default.
      */
-    public function terminate()
+    public function terminateJson(View $view = null)
     {
+        if (!$view) {
+            $view = $this->owner;
+        }
+
         if ($this->canTerminate()) {
-            $this->app->terminateJson($this->owner);
+            $this->app->terminateJson($view);
```

Comment:
Why not
`$this->app->terminateJson($view ?: $this->owner)`?

Reply:
yes, but `??`

Reply:
double improvement :)